tests/common/block_utils.rs

```diff
@@ -287,8 +266,11 @@ pub fn os_hints(
         block_hash: Default::default(),
     };
 
+    // Convert the Blockifier storage into an OS-compatible one
+    let contract_storage_map = build_starknet_storage(&mut blockifier_state);
```

Comment:
Here we need a pre-state as `contract_storage_map` will be used for storage syscalls execution.
See, e.g. https://github.com/starkware-libs/cairo-lang/blob/efa9648f57568aad8f8a13fbf027d2de7c63c2c0/src/starkware/starknet/core/os/execution/execute_syscalls.cairo#L670

Reply:
I'm using the property that `CachedStorage<DictStateReader>` contains both the initial and final states. The initial state remains available through `storage_view` while the final state can be obtained by using `to_state_diff()`.

Reply:
Now I see it.